### PR TITLE
change how to set webhook type and how we default path in builder

### PR DESCRIPTION
**Pull request by mengqiy**
Address comments came from https://github.com/mengqiy/webhook-scaffolding/pull/1

### CODE REVIEW

File: pkg/webhook/admission/builder/builder.go [hunk 1]
```diff
@@ -165,48 +174,48 @@ func (b *WebhookBuilder) Build() (*admission.Webhook, error) {
...
+		}
 	}
 
-	if b.operations == nil {
-		b.operations = []admissionregistrationv1beta1.OperationType{
-			admissionregistrationv1beta1.Create,
-			admissionregistrationv1beta1.Update,
+	if len(b.path) == 0 {
+		if *b.t == types.WebhookTypeMutating {
+			b.path = "/mutating" + w.Rules[0].Resources[0]
```
Comment (DirectXMan12) [inline pkg/webhook/admission/builder/builder.go]: super-nit: maybe just `/mutate-{resource}` or `/validate-{resource}`?

File: pkg/webhook/admission/builder/builder.go [hunk 2]
```diff
@@ -73,10 +73,19 @@ func (b *WebhookBuilder) Name(name string) *WebhookBuilder {
...
+func (b *WebhookBuilder) Mutating() *WebhookBuilder {
+	m := types.WebhookTypeMutating
+	b.t = &m
+	return b
+}
+
+// Validating sets the type to validating admission webhook
+// Only one of Mutating and Validating can be invoked.
+func (b *WebhookBuilder) Validating() *WebhookBuilder {
+	m := types.WebhookTypeValidating
```
Comment (DirectXMan12) [inline pkg/webhook/admission/builder/builder.go]: do we have a default?  Can we have a default?

Comment (mengqiy) [reply pkg/webhook/admission/builder/builder.go]: No, I don't think we can choose a reasonable defaulting for webhook type.

File: example/main.go [hunk 3]
```diff
@@ -76,8 +75,7 @@ func main() {
 	// Setup webhooks
 	mutatingWebhook, err := builder.NewWebhookBuilder().
 		Name("mutating.k8s.io").
-		Type(types.WebhookTypeMutating).
-		Path("/mutating-pods").
+		Mutating().
```
Comment (Liujingfang1) [inline example/main.go]: Maybe change the function names to `BeMutating` and `BeValidating`.

File: example/main.go [hunk 4]
```diff
@@ -103,7 +100,7 @@ func main() {
 	}
 
 	as, err := webhook.NewServer("foo-admission-server", mgr, webhook.ServerOptions{
-		Port:    443,
+		Port:    9876,
```
Comment (Liujingfang1) [inline example/main.go]: Why is this port changed?

File: example/main.go (hunk 3 above)
Comment (mengqiy) [reply example/main.go]: IMHO changing methods names to `BeMutating` and `BeValidating` doesn't buy us much. And just call if `Mutating` is more concise.

**Pull request merged**